yt_dlp/extractor/rai.py

```diff
@@ -183,6 +183,9 @@ def get_format_info(tbr):
         available_qualities = mobj.group('quality').split(',') if mobj.group('quality') else ['*']
         available_qualities = [i for i in available_qualities if i]
 
+        # filter out audio-only formats
+        fmts = [f for f in fmts if not f.get('vcodec') == 'none']
+
```

Comment:
I dont understand this. `fmts` is never used below this point?

Reply:
`ftms` is used inside `get_format_info` inside the for loop just after this line and so I clean the fmts list from audio-only formats one time before the for loop instead of cleaning it every time `get_format_info` is called

Reply:
oh, this makes the code very non-intuitive to read. Move this assignment above the `get_formats_info` function so that it is clearer